### ISSUE TITLE
valid-jsdoc doesn't test exported functions

### ISSUE THREAD

**Issue by matthewwithanm** [eslint/eslint#2522]
The following will cause linter errors (e.g. "Missing JSDoc parameter description", "Use `@return` instead") when using the valid-jsdoc rule:

```js
/**
 * Does…something.
 * @param {string} a
 * @return {Array<number>} The result of doing it
 */
function doSomething(a) {
}
```

However, if you export the function, no errors are raised (i.e. the code passes the linting test unexpectedly):

```js
/**
 * Does…something.
 * @param {string} a
 * @return {Array<number>} The result of doing it
 */
export function doSomething(a) {
}

/**
 * Does something…else.
 * @param {string} a
 * @return {Array<number>} The result of doing it
 */
export default function doSomethingElse(a) {
}
```

My eslintrc doesn't seem to be a factor as I still see the issue when I remove all other rules:

```json
{
  "ecmaFeatures": {
    "modules": true
  },
  "rules": {
    "valid-jsdoc": [2, {
      "prefer": {
        "return": "return"
      }
    }]
  }
}
```

I'm seeing this with 0.19 and 0.21.

**Comment by nzakas** [comment]
What are the errors? Can you paste the output into this issue?

**Comment by matthewwithanm** [comment]
Sorry I wasn't clear. There aren't any errors. The code lints when it shouldn't.

**Comment by matthewwithanm** [comment]
I've updated the issue with some parentheticals to try to clarify that I'm talking about expected validation errors and not unexpected program errors.

**Comment by matthewwithanm** [comment]
Thanks!
